python/langsmith/__init__.py

```diff
@@ -1,17 +1,43 @@
 """LangSmith Client."""
 
-from importlib import metadata
-
-try:
-    __version__ = metadata.version(__package__)
-except metadata.PackageNotFoundError:
-    # Case where package metadata is not available.
-    __version__ = ""
-
-from langsmith.client import Client
-from langsmith.evaluation.evaluator import EvaluationResult, RunEvaluator
-from langsmith.run_helpers import trace, traceable
-from langsmith.run_trees import RunTree
+from typing import Any
+
+
+def __getattr__(name: str) -> Any:
+    if name == "__version__":
```

Comment:
Let's add the imports in a `if TYPE_CHECKING` block as well for a better IDE exp

Reply:
Was there a reason you didn't include the evaluation ones? Can add in a followup (also want to remove dir from below one)

Reply:
Mainly bcs I think it's fine to encourage importing from `langsmith.evaluation` for evals related stuff, but i should probably have the type checking just have it reflect what's available in general